### ISSUE TITLE
animate the sun

### ISSUE THREAD

**Issue by joshmarinacci** [supermedium/aframe-environment-component#14]
Is there a way to animate the sun position?  I've tried moving lightPosition and it seems to update once but not after that.

    <a-entity environment="preset: forest; skyType: atmosphere; ground: hills; lightPosition: 1 5 -3" id="env">
        <a-animation attribute="environment.lightPosition"
                     from="1 0 -3" to="1 5 -3"
                     fill="forwards"
                     repeat="indefinite"
                     dur="10000"
        ></a-animation>
    </a-entity>

**Comment by feiss** [comment]
Mmm... this is strange indeed. `environment.update()` only gets called once for the animation. @ngokevin any clue?

**Comment by dmarcos** [comment]
Can you try with A-Frame master to see if there's any difference?

**Comment by feiss** [comment]
tried. Same... :/

**Comment by joshmarinacci** [comment]
Underneath it creates a separate sunlight entity from the parameters. However I can't seem to animate that either.

**Comment by stevel4857** [comment]
Cool concept.

**Comment by feiss** [comment]
Sorry for the absurd delay. I finally looked into this, and I think there is a problem on using regular `<a-animation>` or `animation` components when animating the sun. I think the most sane option at this time is animating it by using a custom component:

`<a-entity environment="preset: osiris" suntime="from: 0 1 0; to: 1 0 0; duration: 10000"></a-entity>`

```
AFRAME.registerComponent('suntime', {
  schema: {
    from: {type:'vec3', default: {x: 0, y: 1, z: -0.2}},
    to: {type: 'vec3', default: {x: 0, y: 1, z: 0.2}},
    duration: {type: 'int', default: 5000}
  },
  update: function(oldData){
    this.sunPos = {x: this.data.from.x, y: this.data.from.y, z: this.data.from.z};
    this.moveSun();
    this.tween = new AFRAME.TWEEN.Tween(this.sunPos).
      to(this.data.to, this.data.duration).
      easing(TWEEN.Easing.Quadratic.InOut).
      onUpdate(this.moveSun.bind(this)).
      start();
  },
  moveSun: function(){
    var p = this.sunPos;
    this.el.setAttribute('environment', {lightPosition: {x: p.x, y: p.y, z: p.z}});
  }
});
```

**Comment by joshmarinacci** [comment]
No problem. Thanks for the update.

**Comment by vesper8** [comment]
@feiss hey! Thanks a lot for sharing your animation code! It's lovely!!

In the soon-to-be-released 0.9.0 version of Aframe.. it looks like TWEEN has been removed completely and replaced by ANIME

I've been trying to rewrite your TWEEN animation with ANIME but I just can't get it to work.. I was wondering if maybe you could take a look at it and let me know if you see what I'm doing wrong. If I get the answer in the meantime I'll be sure to post it here

Here is what I've got so far

```
      AFRAME.registerComponent('suntime', {
        schema: {
          from: {
            type: 'vec3',
            default: {
              x: 0,
              y: 1,
              z: -0.2,
            },
          },
          to: {
            type: 'vec3',
            default: {
              x: 0,
              y: 1,
              z: 0.2,
            },
          },
          duration: {
            type: 'int',
            default: 5000,
          },
        },
        update() {
          const self = this;

          AFRAME.ANIME({
            targets: this.el,
            translateX: [this.data.from.x, this.data.to.x],
            translateY: [this.data.from.y, this.data.to.y],
            translateZ: [this.data.from.z, this.data.to.z],
            duration: this.data.duration,
            update(anim) {
              self.data.from = {
                x: anim.animations[0].currentValue,
                y: anim.animations[1].currentValue,
                z: anim.animations[2].currentValue,
              };

              self.moveSun(self.data.from);
            },
          });
        },
        moveSun(curPos) {
          this.el.setAttribute('environment', {
            lightPosition: {
              x: curPos.x,
              y: curPos.y,
              z: curPos.z,
            },
          });
        },
      });

```

**Comment by vesper8** [comment]
@feiss hey! Thanks a lot for sharing your animation code! It's lovely!!

In the soon-to-be-released 0.9.0 version of Aframe.. it looks like TWEEN has been removed completely and replaced by ANIME

in case this helps anyone, I got it to work with ANIME with the code below

```
    <a-entity
      environment="preset: osiris"
      sunrise="from: -1 0 0; to: 0 1 0; duration: 5000;"
    ></a-entity>

AFRAME.registerComponent('sunrise', {
        schema: {
          from: {
            type: 'vec3',
            default: {
              x: 0,
              y: 1,
              z: -0.2,
            },
          },
          to: {
            type: 'vec3',
            default: {
              x: 0,
              y: 1,
              z: 0.2,
            },
          },
          duration: {
            type: 'int',
            default: 5000,
          },
        },
        init() {
          this.sunPos = {
            x: this.data.from.x,
            y: this.data.from.y,
            z: this.data.from.z,
          };

          this.moveSun(this.sunPos);
        },
        update() {
          const self = this;

          AFRAME.ANIME({
            targets: this.el,
            translateX: [this.data.from.x, this.data.to.x],
            translateY: [this.data.from.y, this.data.to.y],
            translateZ: [this.data.from.z, this.data.to.z],
            easing: 'easeInQuad',
            duration: this.data.duration,
            update(anim) {
              const curPos = {
                x: anim.animations[0].currentValue,
                y: anim.animations[1].currentValue,
                z: anim.animations[2].currentValue,
              };

              self.moveSun(curPos);
            },
            complete() {
              console.log('sunrise complete!');
            },
          });
        },
        moveSun(curPos) {
          this.el.setAttribute('environment', {
            lightPosition: {
              x: curPos.x,
              y: curPos.y,
              z: curPos.z,
            },
          });
        },
      });
```
